src/makeCurvedLinks.ts

```diff
@@ -30,58 +38,73 @@ const scale = ({ source, target }: Link, size: number) => ({
   },
 })
 
-const addSVGPath = ({ offset = 500, size }: LinksOptions) => (
-  links: Link[],
-) => {
-  return links.map((link: any, i: number, { length }: { length: number }) => {
-    // use size to scale all positions
-    const { source, target } = scale(link, size)
-
-    // don't calcul the curve when only one link between two point
-    if (i === (length - 1) / 2) {
-      return {
-        ...link,
-        d: `M ${source.x} ${source.y} ${target.x} ${target.y}`,
-        quadraticPoint: {
-          x: source.x + (target.x - source.x) / 2,
-          y: source.y + (target.y - source.y) / 2,
-        },
-        sweep: 1,
-      }
+const createCurve = (
+  link: any,
+  i: number,
+  inGroup: number,
+  groups: number,
+  { size, offset = 500 }: LinksOptions,
+): OutputLink => {
+  // use size to scale all positions
+  const { source, target } = scale(link, size)
+
+  // don't calcul the curve when only one link between two point
+  if (i === (inGroup - 1) / 2) {
+    return {
+      ...link,
+      d: `M ${source.x} ${source.y} ${target.x} ${target.y}`,
+      quadraticPoint: {
+        x: source.x + (target.x - source.x) / 2,
+        y: source.y + (target.y - source.y) / 2,
+      },
+      sweep: 1,
     }
+  }
 
-    const cx = (source.x + target.x) / 2
-    const cy = (source.y + target.y) / 2
-    const dx = (target.x - source.x) / 2
-    const dy = (target.y - source.y) / 2
+  const cx = (source.x + target.x) / 2
+  const cy = (source.y + target.y) / 2
+  const dx = (target.x - source.x) / 2
+  const dy = (target.y - source.y) / 2
 
-    const dd = Math.sqrt(dx * dx + dy * dy)
+  const dd = Math.sqrt(dx * dx + dy * dy)
 
-    const sweep = link.source.x - link.target.x > 0 ? 1 : -1
+  const sweep = link.source.x - link.target.x > 0 ? 1 : -1
 
-    const quadraticPoint = {
-      x:
-        cx +
-        (dy / dd) * (offset / links.length) * (i - (length - 1) / 2) * sweep,
-      y:
-        cy -
-        (dx / dd) * (offset / links.length) * (i - (length - 1) / 2) * sweep,
-    }
+  const quadraticPoint = {
+    x: cx + (dy / dd) * (offset / groups) * (i - (inGroup - 1) / 2) * sweep,
+    y: cy - (dx / dd) * (offset / groups) * (i - (inGroup - 1) / 2) * sweep,
+  }
 
-    // add svg path of link
-    return {
-      ...link,
-      d: `M ${source.x} ${source.y} Q ${quadraticPoint.x} ${quadraticPoint.y} ${target.x} ${target.y}`,
-      quadraticPoint,
-      sweep,
-    }
-  })
+  // add svg path of link
+  return {
+    ...link,
+    d: `M ${source.x} ${source.y} Q ${quadraticPoint.x} ${quadraticPoint.y} ${target.x} ${target.y}`,
+    quadraticPoint,
+    sweep,
+  }
 }
 
-export default (links: Link[], { offset, size = 1 }: LinksOptions) => {
+const addSVGPath = (options: LinksOptions) => (groups: Link[]) => {
+  return groups.map((link, index, { length }) =>
+    createCurve(link, index, length, groups.length, options),
+  )
+}
+
+export default (
```

Comment:
the important things is here
the other modification was to understand the algorithm better